### ISSUE TITLE
Control field's value doesn't update in the record properly for controlled values being `null` at some point

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/react-advanced-form#167]
## What
It seems that the changes of `createField()` broke the algorithm of how the controlled fields propagate the value updates to the field record.

## Why
1. `fieldRecord.controllable` was determined by `isset(value)`. That resulted to `false` when handling the following components:
```jsx
<Input name="foo" value={ someVarWhichIsNullAtTheMoment } />
```

2. In `createField.Field.render()` the `fieldProps.value` was simply taken from `props` in case of `fieldState.controllable` and from `fieldState` in case of non-controllable field. While this seems correct, that, again, failed with the example above, as initially `null` value of the `value` prop resulted into this logic to set `fieldProps.value: null`, which excluded `value` from `fieldProps` completely.

3. (Minor) `fieldUtils.validateSync()` always referred to `fieldProps.get('value')`. Although this doesn't cause the described issue, this is a also a caveats and should be covered.

**Comment by kettanaito** [comment]
## Issue
1. There was no logic for dispatching proper onChange handler for controlled fields. That resulted into `Form.handleFieldChange()` being dispatched twice for controlled fields. That caused serious performance issues.

**Comment by kettanaito** [comment]
## Changelog
This was an urgent fix and I didn't issue any pull request to describe the provided changes. I will do this in the issue directly:

1. `onChange` dispatcher for controlled fields was moved to `Form.handleFieldChange()`. The respective `hasChangeHandler` invariant was moved to `Form` level as well. This allowed to simplify the `onChange` handler on `createField.Field` level significantly.
1. Assign `fieldProps.value` properly for controlled fields to propagate to the end `<input>` element
1. Determine if field is controlled not by `isset(value)` but by `props.hasOwnProperty('value')`, which allowed to handle scenarios when `<input value={ nullAtTheMoment } />`

**Comment by kettanaito** [comment]
Merged. Requires testing.

**Comment by kettanaito** [comment]
The issue seems to be resolved.

Published under `0.18.10`. Closing.
